src/main/java/com/mmm/his/cer/utility/farser/ast/parser/AstDescentParser.java

```
@@ -260,12 +262,61 @@ private <X> Expression<C, X> factor(Expression<C, X> left, int leftOperatorPrece
       this.eat(AstCommonTokenType.RPAREN); // Move iterator if 'RPAREN'
     } else if (commonType == AstCommonTokenType.NOT) {
       left = not(left, leftOperatorPrecedence);
+    } else if (commonType == AstCommonTokenType.FUNCTION) {
+      this.eat();
+      left = createNodeWithArgs();
     } else {
       throw new FarserException("Expression malformed on token " + currentToken);
     }
     return left;
   }
 
+  private CommonTokenFlag getCommonTokenFlag() {
+    return currentToken.getType().getCommonTokenType().orElse(null);
+  }
+
+  /**
+   * Verifies that a token is what is expected. If not, an exception is thrown.
+   *
+   * @param expected The expected token.
+   */
+  private void verifyFunctionTokenValidity(AstCommonTokenType expected) {
```

I think I remember that at one point I had implemented `eat(CommonTokenFlag type)` with a check and fail when the type does not match instead of silently continuing and failing later.
Just noting it as a possible option to combine these two.